libs/aws/langchain_aws/chat_models/bedrock.py

```diff
@@ -460,11 +460,11 @@ def _combine_llm_outputs(self, llm_outputs: List[Optional[dict]]) -> dict:
         final_output = {}
         for output in llm_outputs:
             output = output or {}
-            usage = output.get("usage", {})
-            for token_type, token_count in usage.items():
+            token_usage = output.get("usage", {})
+            for token_type, token_count in token_usage.items():
                 final_usage[token_type] += token_count
             final_output.update(output)
-        final_output["usage"] = final_usage
```

Comment:
this is a breaking change.

what we probably want to do is use the new AIMessage.token_usage attribute which is meant to be the standardized way to record token usage: https://github.com/langchain-ai/langchain/pull/21944

Reply:
+1, the standard Langchain is moving toward is the [usage_metadata](https://api.python.langchain.com/en/latest/_modules/langchain_core/messages/ai.html#AIMessage) attribute on AIMessage, which is just a dict with keys `"input_tokens"`, `"output_tokens"`, `"total_tokens"`. If you're able to access the usage data within the `_generate` method, you can construct this dict and then populate `usage_metadata` on the `AIMessage` it generates. Note that we'd need to bump langchain-core >= 0.2.2 in the dependencies.